tests/PHPStan/Analyser/data/bug-8442.php

```diff
@@ -0,0 +1,28 @@
+<?php declare(strict_types = 1);
+
+namespace Bug8442;
+
+use function PHPStan\Testing\assertType;
+use DateInterval;
+
+function () {
+	assertType('false', DateInterval::createFromDateString('foo'));
+	assertType('DateInterval', DateInterval::createFromDateString('1 Day'));
+
+	if (rand(0,1)) {
+		$interval = '1 day';
+	} else {
+		$interval = '2 day';
+	}
+
+	assertType('DateInterval', DateInterval::createFromDateString($interval));
+
+	if (rand(0,1)) {
+		$interval = 'foo';
```

Comment:
those are emitting warnings during testruns, how is that usually dealt with? just `@` the method in the extension?

Reply:
yep, just add the `@` there